test/state/bloom.cpp

```diff
@@ -0,0 +1,64 @@
+// evmone: Fast Ethereum Virtual Machine implementation
+// Copyright 2023 The evmone Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "bloom.hpp"
+#include "state.hpp"
+#include <intx/intx.hpp>
+
+namespace evmone::state
+{
+
+namespace
+{
+/// Adds entry to bloom filter data
+/// based on
+/// https://ethereum.github.io/execution-specs/autoapi/ethereum/shanghai/bloom/index.html?highlight=add_to_bloom#add-to-bloom
+inline void add_to_bloom(bloom& b, const bytes_view& bloom_entry)
+{
+    const auto hash = keccak256(bloom_entry);
+
+    for (const auto i : {0, 2, 4})
+    {
+        const auto bit_to_set = ((hash.bytes[i] & 0x07) << 8) + (hash.bytes[i + 1] & 0xFF);
```

Comment:
I think the documentation from the link above is nice as it explains what supposed to happen briefly, i.e. "take the least significant 11-bits of the first three 16-bit values".